### PR TITLE
arch/risc-v/src/mpfs: Make mpfs_hart_index2id table modifiable by boo…

### PR DESCRIPTION
**Pull request by jlaitine**
…tloader

This is actually the same table as entrypoints, so just use the same data, which can be set before booting any of the harts

### CODE REVIEW

**Review by jpaali — APPROVED** [review]

File: arch/risc-v/src/mpfs/mpfs_opensbi.c [hunk 1]
```diff
@@ -603,6 +580,11 @@ void __attribute__((noreturn)) mpfs_opensbi_setup(void)
 {
   uint32_t hartid = current_hartid();
 
+  for (i = 0; i <  sizeof(mpfs_hart_index2id / sizeof(u32); i++)
```
Comment (jpaali) [inline arch/risc-v/src/mpfs/mpfs_opensbi.c:583]: should be int?

File: arch/risc-v/src/mpfs/mpfs_opensbi.c [hunk 2]
```diff
@@ -185,30 +185,7 @@ static struct aclint_mswi_data mpfs_mswi =
...
-#else
-  [3] = -1,
-#endif
-#ifdef CONFIG_MPFS_HART4_SBI
-  [4] = 4,
-#else
-  [4] = -1,
-#endif
-};
+static u32 mpfs_hart_index2id[MPFS_HART_COUNT];
```
Comment (pussuw) [inline arch/risc-v/src/mpfs/mpfs_opensbi.c:188]: Where the hell is u32 defined. Maybe just use uint32_t like us mere mortals do

File: arch/risc-v/src/mpfs/mpfs_opensbi.c (hunk 1 above)
Comment (pussuw) [reply arch/risc-v/src/mpfs/mpfs_opensbi.c:583]: The / sizeof(u32) should be replaced by / sizeof(mpfs_hart_index2id[0]) so if the type / size changes (it never will though..) this loop will still work

Comment (jlaitine) [reply arch/risc-v/src/mpfs/mpfs_opensbi.c]: What, is someone saying that I should try to compile the code before submitting a PR? Thanks for review, and apologies :smile:

File: arch/risc-v/src/mpfs/mpfs_opensbi.c (hunk 2 above)
Comment (jlaitine) [reply arch/risc-v/src/mpfs/mpfs_opensbi.c:188]: Yes, but the divine beings developing the opensbi have invented this type, and are storing a pointer to this vector....

**Comment by jlaitine** [comment]
Now fixed the issues above, I think

**Review by pussuw — APPROVED** [review]
LGTM

**Pull request merged**